mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -471,15 +471,16 @@ public MessageUid apply(SearchResult input) {
     @Override
     public List<MessageId> search(MailboxSession session, MultimailboxesSearchQuery searchQuery, long limit) throws MailboxException {
         Preconditions.checkArgument(session != null, "'session' is mandatory");
-        return FluentIterable.from(searchMultimap(searchQuery))
-            .transform(new Function<SearchResult, MessageId>() {
-                @Override
-                public MessageId apply(SearchResult input) {
-                    return input.getMessageId().get();
-                }
-            })
-            .limit(Long.valueOf(limit).intValue())
-            .toList();
+        return ImmutableList.copyOf(

```

Comment:
I think limiting before collecting is good because you don't have to make the transform on non used elements.
But I don't see why this has been changed to toSet() instead of toList().